src/DependencyInjection/Configuration.php

```diff
@@ -339,6 +342,133 @@ private function addConnectionsSection(ArrayNodeDefinition $rootNode): void
+                                            ->scalarNode('tlsClientCertificateKeyFilePassword')->end()
+                                            // Local
+                                            ->scalarNode('key')->end()
+                                        ->end()
+                                    ->end()
+                                    ->arrayNode('schemaMap')
+                                        ->prototype('variable')->end()
+                                    ->end()
+                                    ->arrayNode('encryptedFieldsMap')
+                                        ->useAttributeAsKey('name', false)
+                                        ->beforeNormalization()
+                                            ->always(static function ($v) {
+                                                if (is_string($v)) {
+                                                    return json_decode($v, true, 512, JSON_THROW_ON_ERROR);
+                                                }
+
+                                                return $v;
+                                            })->end()
+                                        ->prototype('array')
+                                            ->prototype('array')
+                                                ->children()
+                                                    ->scalarNode('path')->isRequired()->cannotBeEmpty()->end()
+                                                    ->scalarNode('bsonType')->isRequired()->cannotBeEmpty()->end()
+                                                    ->variableNode('keyId')->isRequired()->cannotBeEmpty()->end()
+                                                    ->arrayNode('queries')
+                                                        ->children()
+                                                            ->scalarNode('queryType')->isRequired()->cannotBeEmpty()->end()
+                                                            ->variableNode('min')->end()
+                                                            ->variableNode('max')->end()
+                                                            ->integerNode('sparsity')->end()
+                                                            ->integerNode('precision')->end()
+                                                            ->integerNode('trimFactor')->end()
+                                                            ->integerNode('contention')->end()
+                                                        ->end()
+                                                    ->end()
+                                                ->end()
+                                            ->end()
+                                        ->end()
+                                    ->end()
+                                    ->arrayNode('extraOptions')
+                                        ->prototype('variable')->end()
```

Comment:
You modeled these options in the XSD file. Any reason you chose not to do so here?